api/task.go

```diff
@@ -133,36 +152,42 @@ func (h *taskHandler) updateTask(w http.ResponseWriter, r *http.Request) {
 		if err != nil {
 			err = errors.Wrap(err, fmt.Sprintf("error creating task update"+
 				"event for %q", taskName))
-			log.Printf("[ERROR] (api.task) %s", err)
-			jsonErrorResponse(w, http.StatusInternalServerError, err)
+			logger.Error("error creating new event", "error", err)
+			if err = jsonErrorResponse(w, http.StatusInternalServerError, err); err != nil {
+				logger.Error("error, could not generate json error response", "error", err)
+			}
 			return
 		}
 		defer func() {
 			ev.End(storedErr)
-			log.Printf("[TRACE] (api.task) adding event %s", ev.GoString())
+			logger.Trace("adding event", "event", ev.GoString())
```

Comment:
This is a great pass at updating all of the log lines so my mind is thinking along the next iteration of improving logs. Is there a way to log the event struct as keyed values instead of a string blob? Querying on those event keys could be valuable.

Reply:
I can definitely look into this!

Reply:
I think the easiest thing to do would be a create a function like GoString(), but for KeyValues

```
func (e *Event) KeyValues() []interface{} {
	kv := []interface{"val_1", e.Val1, "val_2", e.Val2}
	return kv
}
```

The only downside is when you pass this to a function you have to be sure to pass it as a variadic number of interfaces rather than a slice:

`logger.Trace("adding event", ev.KeyValues()...)`

Reply:
We can definitely consider doing this in the future, I'll create an issue to track it

Reply:
https://github.com/hashicorp/consul-terraform-sync/issues/377